packages/core/src/fhir-deduplication/resources/diagnostic-report.ts

```diff
@@ -90,26 +93,96 @@ export function groupSameDiagnosticReports(diagReports: DiagnosticReport[]): {
     const datetime = getDateFromResource(diagReport, "datetime");
-    const isPresentedFormPresent = diagReport.presentedForm?.length;
-    const isResultPresent = diagReport.result?.length;
-    if (datetime && (isPresentedFormPresent || isResultPresent)) {
-      const key = JSON.stringify({ datetime });
-      fillMaps(
-        diagReportsMap,
-        key,
-        diagReport,
+    const isPresentedFormPresent = diagReport.presentedForm
+      ? diagReport.presentedForm?.length > 0
+      : false;
+    const isResultPresent = diagReport.result ? diagReport.result?.length > 0 : false;
+
+    // If a diagnostic report does not contain any results or doctor's notes, it is useless and can be removed
+    if (!isPresentedFormPresent && !isResultPresent) {
+      danglingReferences.add(createRef(diagReport));
+      continue;
+    }
+
+    const practitionerRefsSet = new Set<string>();
+
+    const getterKeys: string[] = [];
+    const setterKeys: string[] = [];
+
+    diagReport.performer?.forEach(perf => {
+      const ref = perf.reference;
+      if (ref && ref.includes("Practitioner")) {
+        practitionerRefsSet.add(ref);
+        return;
+      }
+    });
+
+    const practitionerRefs = Array.from(practitionerRefsSet).map(p => ({ practitioner: p }));
+
+    if (isResultPresent) {
+      const resultUuid = createUuidFromText(JSON.stringify(diagReport.result));
+      const key = JSON.stringify({ resultUuid });
+      setterKeys.push(key);
+      getterKeys.push(key);
+    }
+
+    if (isPresentedFormPresent) {
+      const presentedFormUuid = createUuidFromText(JSON.stringify(diagReport.presentedForm));
+      const key = JSON.stringify({ presentedFormUuid });
+      setterKeys.push(key);
+      getterKeys.push(key);
+    }
+
+    if (datetime) {
+      if (practitionerRefs.length > 0) {
+        const practitionerAndDateKeys = createKeysFromObjectArray({ datetime }, practitionerRefs);
+        setterKeys.push(...practitionerAndDateKeys);
+        getterKeys.push(...practitionerAndDateKeys);
+      }
+
+      if (practitionerRefs.length === 0) {
+        const dateKey = JSON.stringify({ datetime });
+        setterKeys.push(dateKey);
+        getterKeys.push(dateKey);
+      }
+    } else {
+      const idKey = JSON.stringify({ id: diagReport.id });
+      setterKeys.push(idKey);
+      getterKeys.push(idKey);
+    }
+
+    if (setterKeys.length > 0) {
+      fillL1L2Maps({
+        map1: l1ReportsMap,
+        map2: l2ReportsMap,
+        getterKeys,
+        setterKeys,
+        targetResource: diagReport,
         refReplacementMap,
-        undefined,
-        removeCodesAndAssignStatus
-      );
+        applySpecialModifications: removeCodesAndAssignStatus,
+      });
     } else {
       danglingReferences.add(createRef(diagReport));
     }
   }
 
   return {
-    diagReportsMap,
+    diagReportsMap: l2ReportsMap,
     refReplacementMap,
     danglingReferences,
   };
 }
+
+function createUuidFromText(input: string) {
```

Comment:
Can we call the result something other than UUID? Those have a pattern, so calling UUID can be misleading if downstream we expect to validate it as such.

Reply:
I believe the output of this function follows that pattern